Schema integrity pipelines are failing because they cannot find an artifact

Engineering DRI: @grlimacan

### Context

<!-- Please, describe the context and need for this task.-->
Schema integrity pipelines are failing because they are trying to download an artifact before it's built.

`[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Optimize Parent 8.6.2-SNAPSHOT:
[INFO] 
[INFO] Optimize Parent .................................... SUCCESS [ 15.223 s]
[INFO] Optimize Client .................................... SUCCESS [  0.598 s]
[INFO] Optimize Util ...................................... SUCCESS [  2.543 s]
[INFO] Optimize Commons ................................... SUCCESS [ 25.522 s]
[INFO] Optimize Backend ................................... SUCCESS [ 37.719 s]
[INFO] Optimize Upgrade ................................... SUCCESS [  3.183 s]
[INFO] Optimize ........................................... SUCCESS [ 16.828 s]
[INFO] optimize-qa ........................................ SUCCESS [  0.560 s]
[INFO] optimize-schema-integrity-tests .................... FAILURE [  4.991 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:45 min (Wall Clock)
[INFO] Finished at: 2024-10-11T12:55:48+02:00
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.8.0:unpack (unpack-previous-and-current-optimize) on project optimize-schema-integrity-tests: Unable to find/resolve artifact. Could not find artifact io.camunda.optimize:camunda-optimize:tar.gz:production:8.6.2-SNAPSHOT in camunda-nexus (https://repository.nexus.camunda.cloud/content/groups/internal/) -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :optimize-schema-integrity-tests
Error: Process completed with exit code 1.`

### Acceptance Criteria

<!-- Please list the Acceptance Criteria. It will be used during Review -->

### Testing Notes

<!-- Any additional information that can be used for review should be provided here -->

### Links

<!-- Add links to related issues -->

### Breakdown

<!-- A breakdown of tasks that need to be completed in order for this to be ready for review. -->
<!--
- [ ] #123
- [ ] Step X
-->

```[tasklist]
### Pull Requests
- [ ] https://github.com/camunda/camunda/pull/23381
- [ ] https://github.com/camunda/camunda/pull/23394
```

## Task Lifecycle

For managing the issue lifecycle, please use the workflow commands. You can see the available
commands by writing `/help` as a comment on this issue.

### Review

#### Engineering Review

- [x] All code targeting the main branch has been reviewed by at least one Engineer
- [x] The PR targeting the main branch has been approved by the reviewing Engineer
- [x] If the API has changed, the API documentation is updated
- [x] All other PRs (docs, controller etc.) in the breakdown have been approved

#### QA Review

- [ ] The change is implemented as described on all target environments/versions/modes
- [ ] The documentation changes are as expected

### Completion

- [x] All Review stages are successfully completed
- [x] All associated PRs are merged to the main branch(es) and stable branches
- [x] The correct version labels are applied to the issue

--- a/.github/workflows/optimize-ci-build-reusable.yml
+++ b/.github/workflows/optimize-ci-build-reusable.yml
@@ -78,7 +78,7 @@ jobs:
       - name: Generate production .tar.gz
         uses: ./.github/actions/run-maven
         with:
-          parameters: -f optimize/pom.xml install -DskipTests -Dskip.docker -PrunAssembly
+          parameters: -f optimize/pom.xml install -DskipTests -Dskip.docker -PrunAssembly -T 1
 
       #########################################################################
       # Build Docker image

--- a/.github/workflows/optimize-ci.yml
+++ b/.github/workflows/optimize-ci.yml
@@ -99,7 +99,7 @@ jobs:
       - name: Install
         uses: ./.github/actions/run-maven
         with:
-          parameters: -f optimize/pom.xml install -Dskip.fe.build -Dskip.docker -DskipTests -PrunAssembly
+          parameters: -f optimize/pom.xml install -Dskip.fe.build -Dskip.docker -DskipTests -PrunAssembly -T 1
       - name: Verify optimize/qa/schema-integrity-tests
         uses: ./.github/actions/run-maven
         with:
@@ -162,7 +162,7 @@ jobs:
       - name: Install
         uses: ./.github/actions/run-maven
         with:
-          parameters: -f optimize/pom.xml install -Dskip.fe.build -Dskip.docker -DskipTests -PrunAssembly
+          parameters: -f optimize/pom.xml install -Dskip.fe.build -Dskip.docker -DskipTests -PrunAssembly -T 1
       - name: Verify optimize/qa/schema-integrity-tests
         uses: ./.github/actions/run-maven
         with:

